File: scripts/reference-solutions/longest-palindrome.py
from collections import Counter
import sys

s = sys.stdin.read()
if s.endswith("\n"):
    s = s[:-1]
if s.endswith("\r"):
    s = s[:-1]

counts = Counter(s)
length = 0
has_odd = False

for count in counts.values():
    length += (count // 2) * 2
    has_odd = has_odd or count % 2 == 1

print(length + (1 if has_odd else 0))

File: scripts/reference-solutions/longest-palindrome.cpp
#include <array>
#include <iostream>
#include <string>

using namespace std;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  string s;
  getline(cin, s, '\0');
  if (!s.empty() && s.back() == '\n') {
    s.pop_back();
  }
  if (!s.empty() && s.back() == '\r') {
    s.pop_back();
  }

  array<int, 256> counts{};
  for (unsigned char c : s) {
    counts[c]++;
  }

  int length = 0;
  bool hasOdd = false;
  for (int count : counts) {
    length += (count / 2) * 2;
    hasOdd = hasOdd || (count % 2 == 1);
  }

  cout << length + (hasOdd ? 1 : 0) << '\n';
  return 0;
}
